File: dorothy/week10/1.py
'''
체력 (i-1) % 4 + 1 까임
체력 0 이하일 때 쓰러짐
권총을 발사해야 하는 횟수
0 1 2 3
1 3 2 4
1 2 3 4
'''
import sys
n = int(sys.stdin.readline().strip())
h = [int(x) for x in sys.stdin.readline().strip().split()]

cnt = 0

for e in h:
	remain = e % 10
	cnt += (e // 10) * 4
	if remain > 0:
		for i in range(4):
			cnt += 1
			remain -= ((cnt-1) % 4) + 1
			if remain <= 0:
				break

print(cnt)

File: leo/week10/1.js
// Run by Node.js
const readline = require("readline");

(async () => {
  let rl = readline.createInterface({ input: process.stdin });

  let input = [];
  for await (const line of rl) {
    input.push(line);
  }
  rl.close();

  const n = parseInt(input[0]);
  const h = input[1].split(" ").map(Number);

  let cnt = 0;

  for (let e of h) {
    let remain = e % 10;
    cnt += Math.floor(e / 10) * 4;
    if (remain > 0) {
      for (let i = 0; i < 4; i++) {
        cnt += 1;
        remain -= ((cnt - 1) % 4) + 1;
        if (remain <= 0) {
          break;
        }
      }
    }
  }

  console.log(cnt);
  process.exit();
})();
